asset_stream_manager/local_assets_stream_manager_service_impl.cc

```diff
@@ -177,6 +160,73 @@ grpc::Status LocalAssetsStreamManagerServiceImpl::StopSession(
   return ToGrpcStatus(status);
 }
 
+absl::Status LocalAssetsStreamManagerServiceImpl::StartSessionInternal(
+    const StartSessionRequest* request, std::string* instance_id,
+    MultiSession** ms, metrics::DeveloperLogEvent* evt) {
+  instance_id->clear();
+  *ms = nullptr;
+  evt->as_manager_data = std::make_unique<metrics::AssetStreamingManagerData>();
+  evt->as_manager_data->session_start_data =
+      std::make_unique<metrics::SessionStartData>();
+  evt->as_manager_data->session_start_data->absl_status = absl::StatusCode::kOk;
+  evt->as_manager_data->session_start_data->status =
+      metrics::SessionStartStatus::kOk;
+  evt->as_manager_data->session_start_data->origin =
+      ConvertOrigin(request->origin());
+
+  if (!(request->gamelet_name().empty() ^ request->user_host().empty())) {
+    return absl::InvalidArgumentError(
+        "Must set either gamelet_name or user_host.");
+  }
+
+  if (request->mount_dir().empty()) {
+    return absl::InvalidArgumentError("mount_dir cannot be empty.");
+  }
+
+  SessionTarget target;
+  target.user_host = request->user_host();
```

Comment:
What do you think about splitting this into two separate functions from here on: `StartSessionStadia` and `StartSessionLinux`? It seems as if the code for both does not have much overlap.

Reply:
Split into GetTargetForStadia and GetTarget (maybe we'll support Windows at some point?).